cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/common/KeyValueListParser.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+
+package io.cdap.cdap.runtime.spi.common;
+
+import com.google.common.base.Splitter;
+import java.util.Iterator;
+import java.util.regex.Pattern;
+
+/**
+ * Parse String values from KeyValue Widgets into individual Key Value entries
+ */
+
+public class KeyValueListParser {
+
+    private final Pattern pairDelimiter;
+    private final Pattern keyValDelimiter;
+    public static final KeyValueListParser DEFAULT = new KeyValueListParser(",", ":");
+
+    public KeyValueListParser(String pairDelimiterRegex, String keyValDelimiterRegex) {
+        this.pairDelimiter = Pattern.compile(pairDelimiterRegex);
+        this.keyValDelimiter = Pattern.compile(keyValDelimiterRegex);
+    }
+
+    public Iterable<KeyValue<String, String>> parse(String kvList) {
```

Comment:
The caller always iterate and insert to a map. Just return `Map<String, String>`, with `LinkedHashMap` as the implementation.

Reply:
I have removed this file entirely and used a different logic.